som_ov_invoices/som_ov_invoices.py

```diff
@@ -65,41 +107,50 @@ def download_invoice_pdf(self, cursor, uid, vat, invoice_number, context=None):
         if context is None:
             context = {}
 
-        users_obj = self.pool.get('som.ov.users')
-        partner = users_obj.get_customer(cursor, uid, vat)
-        invoice_obj = self.pool.get('giscere.facturacio.factura')
-        search_params = [
-            ('number', '=', invoice_number),
-        ]
+        invoice_id = self.validate_invoices(cursor, uid, vat, [invoice_number])
 
-        invoice_id = invoice_obj.search(cursor, uid, search_params)
         if not invoice_id:
             raise NoSuchInvoice(invoice_number)
 
-        invoice = invoice_obj.browse(cursor, uid, invoice_id)[0]
-
-        if invoice.partner_id.id != partner.id:
-            raise UnauthorizedAccess(
-                username=vat,
-                resource_type='Invoice',
-                resource_name=invoice_number,
-            )
-
         report_factura_obj = netsvc.LocalService('report.giscere.factura')
-        result, result_format = report_factura_obj.create(cursor, uid, invoice_id, {})
 
-        filename = (
-            '{invoice_code}_{cil}.pdf'
-        ).format(
-            invoice_code=invoice.number,
-            cil=invoice.cil_id.name,
-        )
+        result, result_formst, filename = self.do_invoice_pdf(cursor, uid, report_factura_obj, invoice_id)
 
         return dict(
             content=base64.b64encode(result),
             filename=filename,
             content_type='application/{}'.format(result_format),
         )
 
+    @www_entry_point(
+        expected_exceptions=(
+            NoSuchUser,
+            UnauthorizedAccess,
+        )
+    )
+    def download_invoices_zip(self, cursor, uid, vat, invoice_numbers, context=None):
+        if context is None:
+            context = {}
+
+        invoice_id = self.validate_invoices(cursor, uid, vat, invoice_numbers)
+
+        report_factura_obj = netsvc.LocalService('report.giscere.factura')
+
+        zipfile_io = StringIO.StringIO()
+        zipfile_ = zipfile.ZipFile(
```

Comment:
Totally opinionated, if there is no sufix as a part of the variable name maybe `_` at `zipfile_`  could be removed